source/common/network/dns_impl.h

```diff
@@ -1,47 +1,68 @@
 #pragma once
 
+#include "envoy/event/file_event.h"
 #include "envoy/network/dns.h"
 
 #include "common/common/linked_object.h"
 #include "common/event/dispatcher_impl.h"
 
+#include "ares.h"
 #include "event2/event_struct.h"
 
 namespace Network {
 
+class DnsResolverImplPeer;
```

Comment:
If it's not easy to refactor or would make the code substantially worse, then I would skip it. It's just not a pattern that we are currently using anywhere. If it's straightforward it would be good to do for consistency reasons.